pagopa/io-app-design-system
# [IOAPPX-326] Fix bug on Android when using `SemiBold` font weight

## Short description
This PR fixes a critical UI bug on Android when using the `SemiBold` font weight. The new _**Titillium Sans Pro**_ uses `Semibold` as a font attribute, while the previous _**Titillium Web**_ fonts used `SemiBold`. Android didn't handle this edge case and rendered everything with the fallback system font.
  
## List of changes proposed in this pull request
- Replace `SemiBold` font weight attribute with `Semibold` in all the typographic styles
- Temporarily update `MDH2` with a different font size to distinguish it from `MDH3`

## How to test
1. Launch the example app with an Android physical or virtual device
2. Check **Typography** screen
